packages/wonder-blocks-dropdown/components/action-menu.js

```diff
@@ -0,0 +1,237 @@
+     */
+    disabled?: boolean,
+    /**
+     * Callback for when the opener is pressed.
+     */
+    onClick: () => void,
+};
+
+class ActionMenuOpener extends React.Component<OpenerProps> {
+    static defaultProps = {
+        disabled: false,
+    };
+
+    render() {
+        const {children, disabled, onClick} = this.props;
+
+        // TODO: incorporate caret once Icon is done
+        return (
+            <Button
+                role="menu"
+                color={"default"}
+                disabled={disabled}
+                kind={"tertiary"}
+                light={false}
+                onClick={onClick}
+                style={[styles.opener]}
+            >
+                {children}
+            </Button>
+        );
+    }
+}
+
+type ItemProps = ActionItemProps | SelectItemProps | SeparatorProps;
+
+type MenuProps = {
+    /**
+     * The items in this menu.
+     */
+    items: Array<ItemProps>,
+
+    /**
+     * Text for the opener of this menu.
+     */
+    menuText: string,
+
+    /**
+     * Whether this menu contains select items. If true, user should also define
+     * the onChange and selectedItems props.
+     */
+    containsSelectionItems?: boolean,
+
+    /**
+     * A callback that returns items that are newly selected. Use only if this
+     * menu contains select items (prop containsSelectionItems).
+     */
+    onChange?: (selectedItems: Array<string>) => void,
+
+    /**
+     * The values of the items that are currently selected. Use only if this
+     * menu contains select items (prop containsSelectionItems).
+     */
+    selectedValues?: Array<string>,
+
+    /**
+     * Whether this menu should be left-aligned or right-aligned with the
+     * opener component. Defaults to left-aligned.
+     */
+    alignment?: "left" | "right",
+
+    /**
+     * Whether this menu is disabled. A disabled menu may not be opened and
+     * does not support interaction. Defaults to false.
+     */
+    disabled?: boolean,
+
+    /**
+     * Optional styling to add.
+     */
+    style?: any,
+};
+
+type State = {
+    /**
+     * Whether or not menu is open.
+     */
+    open: boolean,
+};
+
+export default class ActionMenu extends React.Component<MenuProps, State> {
+    static defaultProps = {
+        alignment: "left",
+        containsSelectionItems: false,
+        disabled: false,
+    };
+
+    constructor(props: MenuProps) {
+        super(props);
+
+        this.state = {
+            open: false,
+        };
+    }
+
+    toggleMenu() {
+        this.setState((prevState) => ({
+            open: !prevState.open,
+        }));
+    }
+
+    handleSelected(selectedValue: string, newSelectionState: boolean) {
+        const {onChange, selectedValues} = this.props;
+
+        // If either of these are not defined, return.
+        if (!onChange || !selectedValues) {
+            return;
+        }
+
+        if (newSelectionState) {
+            onChange([...selectedValues, selectedValue].sort());
```

Comment:
Why sort?  We may even want to switch to `Set` after the React 16 upgrade to make it more explicit that people should not be relying on the order of of this.

Reply:
(The current implementation sorts the list, but yeah there's no reason to do it so I'll remove it for now.)